packages/file_selector/file_selector_macos/lib/file_selector_macos.dart

```diff
@@ -79,13 +79,30 @@ class FileSelectorMacOS extends FileSelectorPlatform {
     String? initialDirectory,
     String? confirmButtonText,
   }) async {
-    return _channel.invokeMethod<String>(
+    final List<String>? pathList = await _channel.invokeListMethod<String>(
+      'getDirectoryPath',
+      <String, dynamic>{
+        'initialDirectory': initialDirectory,
+        'confirmButtonText': confirmButtonText,
+      },
+    );
+    return pathList?.first;
```

Comment:
This and `openFile` are actually both fragile; they are assuming cancel will be null rather than an empty array, which isn't something we should count on.

Let's add a helper function to use in both:
```
String? firstOrNull(List<String>? list) {
  return (list?.isEmpty ?? true) ? null : list.first;
}
```